service/workflow/views.py

```diff
@@ -11,179 +13,93 @@
-                "description": "The requested context does not currently exist."
-            }, status=status.http_404_not_found)
+    queryset = Workflow.objects.all()
+    serializer_class = WorkflowSerializer
 
-        # TODO Use the following line to run an optimized
-        # version of `run` that only runs the affected fns.
-        # Onf faliure here, defer to recreating the available messages
-        #operation = models.Operation.objects.get(pk=kwargs['pk'])
 
+class CaseViewSet(ModelViewSet):
 
-        previous_context = None
-        while not previous_context == context.values: # This is ugly.
-            previous_context = copy.deepcopy(context.values)
-            # Delete all current messages.
-            context.messages.all().delete()
-            context.save()
-
-            # Rebuild all messages from tree.
-            for workflow in context.workflows.all():
-                context 
-                if utils.run(context, user_args, workflow.resolver):
-                    print("WORKFLOW COMPLETED")
-                    context.messages.all().delete()
-                    message = models.Message()
-                    message.message_type = 'Notification'
-                    message.response = None
-                    message.content = 'Thank you for completing the workflow.'
-                    message.ctx = context
-                    message.save()
-
-        return Response("Success")
-
+    queryset = Case.objects.all()
+    serializer_class = CaseSerializer
 
+    def perform_create(self, serializer):
+        case = serializer.save()
+        case.workflow.update(case)
 
-class TransitionRelationship(RelationshipView):
-    queryset = models.Transition.objects.all()
-    serializer_class = models.Transition
 
+class ActionViewSet(ModelViewSet):
 
-class Location(viewsets.ModelViewSet):
-    queryset = models.Location.objects.all()
-    serializer_class = serializers.Location
+    queryset = Action.objects.all()
+    serializer_class = ActionSerializer
 
 
-class Arc(viewsets.ModelViewSet):
-    queryset = models.Arc.objects.all()
-    serializer_class = serializers.Arc
+class ParameterViewSet(ModelViewSet):
 
+    queryset = Parameter.objects.all()
+    serializer_class = ParameterSerializer
 
-class Token(viewsets.ModelViewSet):
-    queryset = models.Token.objects.all()
-    serializer_class = serializers.Token
 
-    def perform_create(self, serializer):
+class ConditionViewSet(ModelViewSet):
 
-        token = serializer.save()
-        token.net.wake()
+    queryset = Condition.objects.all()
+    serializer_class = ConditionSerializer
 
 
-class Case(viewsets.ModelViewSet):
-    queryset = models.Case.objects.all()
-    serializer_class = serializers.Case
+class MessageViewSet(ModelViewSet):
 
-    def perform_create(self, serializer):
+    queryset = Message.objects.all()
+    serializer_class = MessageSerializer
 
-        case = serializer.save()
+    def get_queryset(self):
```

Comment:
lets the frontend filter messages by case so it can show only messages for one case in the submission form